dali/operators/sequence/optical_flow/optical_flow.h

```diff
@@ -79,11 +81,50 @@ class OpticalFlow : public Operator<Backend> {
 
  protected:
   bool SetupImpl(std::vector<OutputDesc> &output_desc, const workspace_t<Backend> &ws) override {
-    return false;
+    const auto &input = ws.template Input<Backend>(0);
+    if (enable_external_hints_) {
+      const auto &hints = ws.template Input<Backend>(1);
+      // Extract calculation params
+      ExtractParams(input, hints);
+    } else {
+      ExtractParams(input);
+    }
+
+    cudaStream_t of_stream = ws.stream();
+    #if NVML_ENABLED
+      {
+        static float driver_version = nvml::GetDriverVersion();
+        if (driver_version > 460 && driver_version < 470.21)
+          of_stream = 0;
+      }
+    #else
+      {
+        int driver_cuda_version = 0;
+        CUDA_CALL(cuDriverGetVersion(&driver_cuda_version));
+        if (driver_cuda_version >= 11030 && driver_cuda_version < 11040)
+          of_stream = 0;
+      }
+    #endif
+
+    auto input_sh = input.shape();
+    of_lazy_init(input_sh[0][2], input_sh[0][1], depth_, image_type_, device_id_, of_stream);
```

Comment:
If `std::vector<DimsOrder> processing_order;` was created here, we could make sure this gets initialized with the actual shape of first sample, right?

Reply:
I can do it, but it will impact the perf only of the very first iteration as this function internals are called only once for the operators lifetime.

Reply:
Done